purchase_request/models/purchase_request.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+_STATES = [
+    ('draft', 'Draft'),
+    ('to_approve', 'To be approved'),
+    ('approved', 'Approved'),
+    ('rejected', 'Rejected')
+]
+
+
+class PurchaseRequest(models.Model):
+
+    _name = 'purchase.request'
+    _description = 'Purchase Request'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    @api.model
+    def _company_get(self):
+        company_id = self.env['res.company']._company_default_get(self._name)
+        return self.env['res.company'].browse(company_id.id)
+
+    @api.model
+    def _get_default_requested_by(self):
+        return self.env['res.users'].browse(self.env.uid)
+
+    @api.model
+    def _get_default_name(self):
+        return self.env['ir.sequence'].get('purchase.request')
```

Comment:
@adrienpeiffer This is the standard approach in other parts of Odoo. We adopted as much as possible the best practices. If a number is lost in the sequence, so be it. Only in invoices it can really a problem to have a gap in the invoice sequence.

In general, as this module has already been merged in 8.0, I suggest not to change the behaviour in 9.0, unless it is really mandatory by the change of design in 9.0.